base/array.jl

```diff
@@ -255,6 +255,7 @@ end
 # Deprecate this, as it seems to have no documented meaning and is unused here,
 # but is frequently accessed in packages
 elsize(@nospecialize _::Type{A}) where {T,A<:Array{T}} = aligned_sizeof(T)
+elsize(@nospecialize T::Type{<:Ptr}) = aligned_sizeof(eltype(T))
```

Comment:
As I said before, you need to use the correct elsize definition: https://github.com/JuliaLang/julia/pull/49811/files/17540efcbe1d3347c66125c1275dc4e7b755cb61#r1199658397
> We need to use a definition of `aligned_sizeof` that actually corresponds to what the intrinsics expect.
> 
> ```julia
> function elsize(::Type{Ptr{T}}) where T
>     # this only must return something valid for values which satisfy is_valid_intrinsic_elptr(T),
>     # which includes Any and most concrete datatypes
>     T === Any && return sizeof(Ptr{Any})
>     T isa DataType || sizeof(Any) # throws
>     return LLT_ALIGN(Core.sizeof(T), datatype_alignment(T))
> end
> ```